### PR TITLE
[CORL-1029] Move to Squire and more

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- **\*breaking\*** Move to [squire-rte](https://github.com/neilj/Squire) to do the low level work and act as our feature API 
- **\*breaking\*** Remove `text` from `onChange`. Now `onChange` only returns the `html` value.
- **\*breaking\*** Rename `inputId` to `inputID`
- Add `Strike`, `OrderedList`, `UnorderedList`, and `Spoiler` feature
- Allow pasting html input
- Allow custom html sanitization 
- Upgrade dependencies

### CODE REVIEW

File: src/RTE.tsx [hunk 1]
```diff
@@ -1,63 +1,98 @@
...
   onFocus?: EventHandler<FocusEvent>;
+  /** onFocus is called whenenver the RTE looses focus */
   onBlur?: EventHandler<FocusEvent>;
+  /** Only allow pasting text */
+  pasteTextOnly?: boolean;
+  /**
+   * If true, don't sanitize when `value` is applied.
+   * Important: No sanitization will take place unless `sanitizeToDOMFragment` is set.
+   */
+  noSanitizeValue?: boolean;
```
Comment (tessalt) [inline src/RTE.tsx]: the naming of these is a bit inconsistent in that they're negatives... could it be `sanitizeValue` and by default set it to false?

**Review by tessalt — CHANGES_REQUESTED** [review]
Looks great! just one note about confusing prop naming

File: src/RTE.tsx (hunk 1 above)
Comment (alinhle) [reply src/RTE.tsx]: You mean default to `true`. Yeah makes sense!

Comment (alinhle) [reply src/RTE.tsx]: I removed  `noSanitizeInsert` cause it should never be changed for security reasons and renamed  `noSanitizeValue` to `sanitizeValue`. Thanks @tessalt!

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**